modules/identity-shiro/src/main/java/org/opensearch/identity/shiro/ShiroSubject.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity.shiro;
+
+import java.security.Principal;
+import java.util.Objects;
+
+import org.opensearch.identity.Subject;
+import org.opensearch.identity.tokens.AuthToken;
+
+/**
+ * Subject backed by Shiro
+ *
+ * @opensearch.experimental
+ */
+public class ShiroSubject implements Subject {
+    private final AuthTokenHandler authTokenHandler;
+    private final org.apache.shiro.subject.Subject shiroSubject;
+
+    public ShiroSubject(final AuthTokenHandler authTokenHandler, final org.apache.shiro.subject.Subject subject) {
+        this.authTokenHandler = authTokenHandler;
```

Comment:
Isn't `authTokenHandler` for all subjects the same for OpenSearch? 
Does it have to be a member for each subject?